src/database/mock.rs

```diff
@@ -162,45 +156,47 @@ impl MockDatabaseTrait for MockDatabase {
                 Err(err) => Err(err),
             }
         } else {
-            Err(DbErr::Query(RuntimeErr::Internal(
-                "`query_results` buffer is empty.".to_owned(),
-            )))
+            Err(query_err("`query_results` buffer is empty."))
         }
     }
 
     #[instrument(level = "trace")]
     fn begin(&mut self) {
-        if self.transaction.is_some() {
-            self.transaction
-                .as_mut()
-                .unwrap()
-                .begin_nested(self.db_backend);
-        } else {
-            self.transaction = Some(OpenTransaction::init());
+        match self.transaction.as_mut() {
+            Some(transaction) => transaction.begin_nested(self.db_backend),
+            None => self.transaction = Some(OpenTransaction::init()),
         }
     }
 
     #[instrument(level = "trace")]
     fn commit(&mut self) {
-        if self.transaction.is_some() {
-            if self.transaction.as_mut().unwrap().commit(self.db_backend) {
-                let transaction = self.transaction.take().unwrap();
-                self.transaction_log.push(transaction.into_transaction());
+        match self.transaction.as_mut() {
+            Some(transaction) => {
+                if transaction.commit(self.db_backend) {
+                    let transaction = self
+                        .transaction
+                        .take()
+                        .expect("There is no open transaction to commit");
```

Comment:
I am sure this is unreachable, as it is already checked to be Some

Reply:
How about this? https://github.com/SeaQL/sea-orm/pull/1231/commits/0954fab7a1932ed064f2591ce7dc69143edebd5e

Reply:
Much better